### ISSUE TITLE
Sandstorm: Focused "outline" button does not have background color

### ISSUE THREAD

**Issue by carolinan** [WordPress/twentytwentyfour#569]
**Description**
This may be a non issue. It is not an accessibility issue but I want to mention it because the styles are inconsistent.

Variation: Sandstorm
Button style: Outline

When you hover over a outlined button, the background and text color changes.
When you focus on the outlined button, the only visible difference is that the outline is added.

In comparison, when you focus on the filled button, it does change background and text color, and the outline is added.

**Step-by-step reproduction instructions**
Add two buttons, with links. One should use the default filled style, the other outline.

**Expected behavior**
For the hover and focus styles to be more consistent(?)
Like I said this might be a non-issue. I want to mention it in case it is not intentional.

**Comment by carolinan** [comment]
I am seeing the same on Onyx, Mint.
Rust has the same inconsistency but also no hover style for the filled button.

**Comment by luminuu** [comment]
@beafialho Can you give some feedback/suggestions on this?

**Comment by beafialho** [comment]
Good catch. I tried to exemplify on Figma how these should look across variations. Is this clear?

<img width="1022" alt="Captura de ecrã 2023-10-11, às 12 05 48" src="https://github.com/WordPress/twentytwentyfour/assets/65220155/01d6bfb0-c131-40f7-a424-41f36768dc5f">

**Comment by luminuu** [comment]
@beafialho Can I assume you want the buttons to always:
- On hover, use `contrast-2` as background color for the filled buttons, and text and outline color for the outlined buttons
- On Focus, use the same as above with the defined outline settings

I'll add the current status with screencasts in a follow up comment, so it will be easier to see where work needs to be done.

**Comment by luminuu** [comment]
**Default, Ice, Mint:**

https://github.com/WordPress/twentytwentyfour/assets/7585600/58ff6c96-de1b-427d-9e2d-c9101c078c61


Uses `contrast` (black) as filled background/outline color and `contrast-2` (grey) as hover state background/outline color. For the outlined button, the focus state is incorrect and needs to be `contrast-2` too.

**Fossil:**

https://github.com/WordPress/twentytwentyfour/assets/7585600/17d61d66-5802-4170-b339-bd5c6cf6d222

Uses `contrast-2` as default color and `contrast` as hover color. Since the `focus` state is not explicitly defined, it uses again `contrast-2` from the default style variation for the focus state. Same for the outlined style variation.

**Onyx:**

https://github.com/WordPress/twentytwentyfour/assets/7585600/67c8ef57-ffc7-4ab8-a0d0-3688e8d691db

Basically the same as the default, just with different colors.

**Rust:**

https://github.com/WordPress/twentytwentyfour/assets/7585600/8d410251-e595-4172-b1de-efe2c0ef707f

Since this style variation does not have a `contrast-2` color, `contrast` (the dark orange) is used. Which color should be used as `contrast-2` that can also be added to the general palette? There are only three colors available with Rust at the moment.

**Sanstorm:**

https://github.com/WordPress/twentytwentyfour/assets/7585600/3563b2eb-961c-4b64-b519-a4dc7dacaf4d

This style variation uses almost exactly the same settings as the Default, with only the text color changed to `base-2` in the normal button state. On hover/focus states, the default `base` color is used for the text.

**Comment by beafialho** [comment]
@luminuu thank you! In fact, I had the outlined buttons still be outlined on hover, but I think the fill on hover works better so I'll update the Figma file 👍🏼 

My only comment is please adjust the filled and outline buttons in all variations to always have the same height, which isn't happening at the moment:

<img width="1312" alt="Captura de ecrã 2023-10-12, às 15 11 34" src="https://github.com/WordPress/twentytwentyfour/assets/65220155/fab0d128-7668-451e-aa36-7ace99a0ef5e">

**Comment by luminuu** [comment]
@beafialho I want to point out that it is currently impossible in the editor to define another set of colors for the outlined variation in the `focus` state. The focus border is added, but colors can't change, as shown in all above examples. This is a limitation from the editor. @carolinan pointed this out to me on Twitter earlier.

Do you want to have the buttons in Fossil and Sandstorm be the original size like the outline button, or the bigger size like the default button?

Also, can you create a second color for the Rust variation for the hover states? Currently, there's no hover effect on the normal button, as it defaults to the same background color.

**Comment by beafialho** [comment]
> @beafialho I want to point out that it is currently impossible in the editor to define another set of colors for the outlined variation in the focus state.

Ok, thanks for letting me know 👍🏼 

> Do you want to have the buttons in Fossil and Sandstorm be the original size like the outline button, or the bigger size like the default button?

We should have the buttons be the original size like the outline.

@richtabor originally created the Rust variation, we may want to get his input about it.

**Comment by richtabor** [comment]
I think we should keep varying button styles/radii/colors/padding/etc on variations — probably introduce even more variation across the styles. 

The problem is not an issue with the padding itself, it's that the padding is omitted by core's padding of the outline button.

**Comment by MaggieCabrera** [comment]
I think we ended up fixing this in a different PR (https://github.com/WordPress/twentytwentyfour/pull/649 maybe?) Closing this
